From ab94c0fed1813eb5e8376be51a7c93164652e26b Mon Sep 17 00:00:00 2001
From: Andreas Dangel <andreas.dangel@pmd-code.org>
Date: Thu, 4 May 2023 19:44:31 +0200
Subject: [PATCH] test case 1 - single java rule changed

A single rule (java class) is changed. Only this rule should be executed
and only this rule should be compared (ruleset is filtered).

The rule "AbstractClassWithoutAbstractMethod" doesn't report
any violations anymore, so false-negatives should appear for
exactly this rule.
---
 .../bestpractices/AbstractClassWithoutAbstractMethodRule.java  | 2 +-
 .../bestpractices/AbstractClassWithoutAbstractMethodTest.java  | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodRule.java b/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodRule.java
index 0d0d8c33e4..972e1bd62a 100644
--- a/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodRule.java
+++ b/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodRule.java
@@ -22,7 +22,7 @@ public class AbstractClassWithoutAbstractMethodRule extends AbstractJavaRulechai
         }
 
         if (node.getDeclarations(ASTMethodDeclaration.class).none(ASTMethodDeclaration::isAbstract)) {
-            addViolation(data, node);
+            //addViolation(data, node);
         }
         return data;
     }
diff --git a/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodTest.java b/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodTest.java
index b319c5e9f1..77698edb60 100644
--- a/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodTest.java
+++ b/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AbstractClassWithoutAbstractMethodTest.java
@@ -4,8 +4,11 @@
 
 package net.sourceforge.pmd.lang.java.rule.bestpractices;
 
+import org.junit.jupiter.api.Disabled;
+
 import net.sourceforge.pmd.testframework.PmdRuleTst;
 
+@Disabled
 class AbstractClassWithoutAbstractMethodTest extends PmdRuleTst {
     // no additional unit tests
 }
-- 
2.39.2
